test/v1alpha1/knativekafka.go

```diff
@@ -0,0 +1,106 @@
+package v1alpha1
+
+import (
+	kafkav1alpha1 "github.com/openshift-knative/serverless-operator/knative-operator/pkg/apis/operator/v1alpha1"
+	"github.com/openshift-knative/serverless-operator/test"
+	"github.com/pkg/errors"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func KnativeKafka(name, namespace string) *kafkav1alpha1.KnativeKafka {
+	return &kafkav1alpha1.KnativeKafka{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: kafkav1alpha1.KnativeKafkaSpec{
+			Source: kafkav1alpha1.Source{
+				Enabled: true,
+			},
+			Channel: kafkav1alpha1.Channel{
+				Enabled:          true,
+				BootstrapServers: "my-cluster-kafka-bootstrap.kafka:9092",
+			},
+		},
+	}
+}
+
+func WithKnativeKafkaReady(ctx *test.Context, name, namespace string) (*kafkav1alpha1.KnativeKafka, error) {
+	kafka, err := CreateKnativeKafka(ctx, name, namespace)
+	if err != nil {
+		return nil, err
+	}
+	if _, err = WaitForKnativeKafkaState(ctx, kafka.Name, kafka.Namespace, IsKnativeKafkaReady); err != nil {
+		return nil, err
+	}
+	return kafka, nil
+}
+
+func CreateKnativeKafka(ctx *test.Context, name, namespace string) (*kafkav1alpha1.KnativeKafka, error) {
+	uo, err := runtime.DefaultUnstructuredConverter.ToUnstructured(KnativeKafka(name, namespace))
+	if err != nil {
+		return nil, err
+	}
+	u := &unstructured.Unstructured{Object: uo}
```

Comment:
Ugh.. Is this because we don't have a client generated for KnativeKafka?

Reply:
yeah. I needed to learn all these ugly crap that's handled by operator-sdk OR knative-pkg